catalog/spatial/registry/registry-source-configuration-handler/src/main/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandler.java

```diff
@@ -0,0 +1,700 @@
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.schemabindings.EbrimConstants;
+import org.codice.ddf.registry.schemabindings.helper.RegistryPackageTypeHelper;
+import org.codice.ddf.registry.schemabindings.helper.SlotTypeHelper;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.osgi.service.metatype.AttributeDefinition;
+import org.osgi.service.metatype.MetaTypeInformation;
+import org.osgi.service.metatype.MetaTypeService;
+import org.osgi.service.metatype.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ServiceBindingType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+
+public class SourceConfigurationHandler implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceConfigurationHandler.class);
+
+    private static final String METACARD_PROPERTY = "ddf.catalog.event.metacard";
+
+    private static final String BINDING_TYPE = "bindingType";
+
+    private static final String DISABLED_CONFIGURATION_SUFFIX = "_disabled";
+
+    private static final String ID = "id";
+
+    private static final String SHORTNAME = "shortname";
+
+    private static final String CONFIGURATION_FILTER =
+            "(&(%s=%s)(|(service.factoryPid=*source*)(service.factoryPid=*Source*)(service.factoryPid=*service*)(service.factoryPid=*Service*)))";
+
+    private static final String CREATED_TOPIC = "ddf/catalog/event/CREATED";
+
+    private static final String UPDATED_TOPIC = "ddf/catalog/event/UPDATED";
+
+    private static final String DELETED_TOPIC = "ddf/catalog/event/DELETED";
+
+    private static final int SHUTDOWN_TIMEOUT_SECONDS = 60;
+
+    private ConfigurationAdmin configurationAdmin;
+
+    private FederationAdminService federationAdminService;
+
+    private MetaTypeService metaTypeService;
+
+    private Parser parser;
+
+    private ParserConfigurator unmarshalConfigurator;
+
+    private ExecutorService executor;
+
+    private String urlBindingName;
+
+    private Map<String, String> bindingTypeToFactoryPidMap = new HashMap<>();
+
+    private List<String> sourceActivationPriorityOrder = new ArrayList<>();
+
+    private Boolean activateConfigurations;
```

Comment:
Can we use simple `boolean` types for these? Or even better, any way we can use separate classes or something to replace those flags?